Revit_Core_Engine/Convert/Geometry/ToRevit/Curve.cs

```diff
@@ -77,22 +77,36 @@ public static Curve ToRevit(this oM.Geometry.NurbsCurve curve)
             knots.Insert(0, knots[0]);
             knots.Add(knots[knots.Count - 1]);
             List<XYZ> controlPoints = curve.ControlPoints.Select(x => x.ToRevit()).ToList();
-            
-            try
-            {
-                return NurbSpline.CreateCurve(curve.Degree(), knots, controlPoints, curve.Weights);
-            }
-            catch
+
+            int degree = curve.Degree();
+            switch (degree)
             {
-                BH.Engine.Reflection.Compute.RecordWarning("Conversion of BHoM nurbs curve to Revit based on degree and knot vector failed. A simplified (possibly different) spline has been created.");
-                return NurbSpline.CreateCurve(controlPoints, curve.Weights);
+                case 1:
+                    return new oM.Geometry.Polyline { ControlPoints = curve.ControlPoints }.ToRevit();
+                case 2:
+                {
+                    BH.Engine.Reflection.Compute.RecordWarning("Revit does not support nurbs curves of degree 2. A simplified (possibly distorted) hermite spline has been created.");
+
+                    List<oM.Geometry.Point> cps = new List<oM.Geometry.Point>();
+                    for (int i = 0; i < curve.ControlPoints.Count; i++)
+                    {
+                        if (Math.Abs(1 - curve.Weights[i]) <= 1e-6)
```

Comment:
Tolerance shouldn't be hard coded #536

Reply:
Agreed - would you be happy with merging this PR and adding a todo item to #536 so that this did not get forgotten?